Lib/ntpath.py

```diff
@@ -554,9 +567,20 @@ def _getfinalpathname_nonstrict(path):
         except OSError:
             pass
 
-        # Allow file (2) or directory (3) not found, incorrect parameter (87),
-        # invalid syntax (123), and symlinks that cannot be followed (1921)
-        allowed_winerror = 2, 3, 87, 123, 1921
+        # These error codes indicate that we should stop resolving the path
+        # and return the value we currently have.
+        # 1: ERROR_INVALID_FUNCTION
+        # 2: ERROR_FILE_NOT_FOUND
+        # 3: ERROR_DIRECTORY_NOT_FOUND
+        # 5: ERROR_ACCESS_DENIED
+        # 21: ERROR_NOT_READY (implies drive with no media)
+        # 32: ERROR_SHARING_VIOLATION
```

Comment:
Getting a sharing violation is highly unusual in this case, so I'd include a parenthetical note such as "(implies an NTFS paging file)". 

If I recall correctly, FAT32 fails the paging-file case with a more sensible `STATUS_ACCESS_DENIED` instead of `STATUS_SHARING_VIOLATION`. Normally, a sharing violation error can only occur if we request data access (i.e. read/execute, write/append, delete) on a volume device or a file-system file or directory and the desired access isn't shared by existing opens. We never request data access in these cases, so NTFS returning a sharing violation is unexpected. (Well, since it has behaved like this for over 25 years, it's not unexpected historically, just unexpected for a non-data `CreateFileW` call to fail with this error.)

Anyway, what we do in `os.stat` for `ERROR_ACCESS_DENIED` and `ERROR_SHARING_VIOLATION` is to query the parent directory for basic stat info via `FindFirstFileW`. Do you want to try to get the normalized filename from the parent directory in this case -- e.g. "PAGEFILE.SYS" -> "pagefile.sys", or "C:\Users\JOHNSM\~1" -> "C:\Users\JohnSmith"? We'd need a private function for this such as `nt._get_normalized_name`. In particular, we may have a short alias such as "JOHNSM\~1", which requires `FindFirstFileW` to map it to the real name "JohnSmith". Either name is 'real', but by default `GetFinalPathNameByHandleW` tries to get a normalized name, so we'd be more consistent with its behavior in general.

Reply:
I'll update the comment now, but I think we can leave the additional check for a separate bug.